test-utils/test/shared/act.test.js

```diff
@@ -516,4 +516,56 @@ describe('act', () => {
 			});
 		});
 	});
+
+	describe('act function with finish implementations', function () {
+		let actDepth = 0;
+		beforeEach(function () {
+			actDepth = 0; // Reset actDepth before each test
```

Comment:
Nit: all of the comments you added in this file can be removed. It's very straight-forward code, it speaks for itself.